Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHandDataProvider.cs

```diff
@@ -185,8 +188,16 @@ public bool UpdateHandData(SimulatedHandData handDataLeft, SimulatedHandData han
             // TODO: DateTime.UtcNow can be quite imprecise, better use Stopwatch.GetTimestamp
             // https://stackoverflow.com/questions/2143140/c-sharp-datetime-now-precision
             long timestamp = DateTime.UtcNow.Ticks;
-            handDataChanged |= handDataLeft.UpdateWithTimestamp(timestamp, HandStateLeft.IsTracked, HandStateLeft.IsPinching, HandStateLeft.FillCurrentFrame);
-            handDataChanged |= handDataRight.UpdateWithTimestamp(timestamp, HandStateRight.IsTracked, HandStateRight.IsPinching, HandStateRight.FillCurrentFrame);
+
+            // Cache the generator delegates so we don't cg alloc every frame
+            if (generatorLeft == null)
+                generatorLeft = HandStateLeft.FillCurrentFrame;
```

Comment:
please add { } around these if blocks. too easy to add an indented line that looks to be in the block but isnt.

thanks!